### PR TITLE
Fix/gp interface

### PR DESCRIPTION
**Pull request by edaub**
Fixes a few issues with the GP/MOGP interface. Fixes an issue with the priors list (now uses `None` instead of an empty list) and allows the kernel to be specified via a string. Addresses issue #102.

Changes:

* `GaussianProcess` and `MultiOutputGP` can accept strings for the kernel argument. Priors default to `None` to represent weak prior information
* Updates to documentation to reflect this

### CODE REVIEW

File: mogp_emulator/GaussianProcess.py [hunk 1]
```diff
@@ -86,9 +86,9 @@ def __init__(self, inputs, targets, mean=None, kernel=SquaredExponential(), prio
         ``targets`` is the target data to be fit by the emulator, also held in an array-like
         object. This must be a 1D array of length ``n``.
 
-        ``prior`` must be either (1) an empty list, indicating uninformative prior information,
-        or (2) a list of length ``n_params`` containing either ``None`` (for uninformative prior)
-        or a ``Prior``-derived object.
+        ``prior`` must be either (1) ``None`` or (2) an empty list, indicating uninformative prior
+        information, or (3) a list of length ``n_params`` containing either ``None``-types
+        (for uninformative prior) or a series of ``Prior``-derived objects.
```
Comment (ots22) [inline mogp_emulator/GaussianProcess.py:91]: What is the motivation for accepting both `None` and `[]` to indicate an uninformative prior?  It took me a moment to realise that there is no difference between these cases, and it (slightly) complicates the logic in the `priors` setter below.  Is it to avoid using a mutable default argument while remaining backwards compatible?  How about rewording this to something like the suggested?
```suggestion
        ``prior`` must be a list of length ``n_params`` whose elements are either ``Prior``-derived 
        objects or ``None``.  Each element is used as the prior for the corresponding parameter (with 
        ``None`` indicating the uninformative prior).  Passing the empty list or ``None`` as this 
        argument (in its entirety) may be used as an abbreviation for a list of ``n_params`` lots of 
        ``None``.
```

Comment (edaub) [reply mogp_emulator/GaussianProcess.py:91]: You are correct, I changed this to avoid potential problems with mutable defaut arguments and maintain backwards compatibility (I had not run into any problems using it in practice, but caught this and worried that there might be some case where it caused problems that I hadn't thought of). Your rewrite looks like an improvement, so I will update before merging.
